core/trino-main/src/main/java/io/trino/execution/DropNotNullContraintTask.java

```diff
@@ -0,0 +1,108 @@
+import static io.trino.metadata.MetadataUtil.createQualifiedObjectName;
+import static io.trino.spi.StandardErrorCode.COLUMN_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.spi.connector.ConnectorCapabilities.NOT_NULL_COLUMN_CONSTRAINT;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static java.util.Objects.requireNonNull;
+
+public class DropNotNullContraintTask
+        implements DataDefinitionTask<DropNotNullConstraint>
+{
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public DropNotNullContraintTask(Metadata metadata, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "DROP NOT NULL";
+    }
+
+    @Override
+    public ListenableFuture<Void> execute(
+            DropNotNullConstraint statement,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
+    {
+        Session session = stateMachine.getSession();
+        QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTable());
+        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, tableName);
+        if (redirectionAwareTableHandle.tableHandle().isEmpty()) {
+            String exceptionMessage = "Table '%s' does not exist".formatted(tableName);
+            if (metadata.getMaterializedView(session, tableName).isPresent()) {
+                exceptionMessage += ", but a materialized view with that name exists.";
+            }
+            else if (metadata.getView(session, tableName).isPresent()) {
+                exceptionMessage += ", but a view with that name exists.";
+            }
+            if (!statement.isTableExists()) {
+                throw semanticException(TABLE_NOT_FOUND, statement, "%s", exceptionMessage);
+            }
+            return immediateVoidFuture();
+        }
+        accessControl.checkCanAlterColumn(session.toSecurityContext(), tableName);
+
+        TableHandle tableHandle = redirectionAwareTableHandle.tableHandle().get();
+        CatalogHandle catalogHandle = tableHandle.getCatalogHandle();
+        String column = statement.getColumn().toString();
+        ColumnHandle columnHandle = metadata.getColumnHandles(session, tableHandle).get(column);
+
+        if (!metadata.getConnectorCapabilities(session, catalogHandle).contains(NOT_NULL_COLUMN_CONSTRAINT)) {
+            throw semanticException(NOT_SUPPORTED, statement, "Catalog '%s' does not support NOT NULL for column '%s'", catalogHandle, column);
+        }
```

Comment:
Why do we need this? If the connector doesn't support dropNotNullConstraint, it should fail the call below:

> `metadata.dropNotNullConstraint(session, tableHandle, columnHandle);`

Having two mechanisms to determine whether the functionality is supported is going to lead to implementations getting out of sync, developers not knowing where to implement that validation, etc.